p2p/swarmhandlers.go

```diff
@@ -74,13 +74,16 @@ func (s *swarmImpl) onConnectionRequest(req node.RemoteNodeData, done chan error
 
 	var err error
 
+	s.peerMapMutex.Lock()
+
 	// check for existing session with that node
 	peer := s.peers[req.ID()]
-
+	s.peerMapMutex.Unlock()
 	if peer == nil {
 		s.onRegisterNodeRequest(req)
 	}
-
+	s.peerMapMutex.Lock()
+	defer s.peerMapMutex.Unlock()
```

Comment:
sure we can't unlock it earlier ? the mutex protects the peer map. but the peer state is protected by another mutex so if the peer map isn't touched we can release the mutex.